nautobot/dcim/models/devices.py

```diff
@@ -818,21 +818,20 @@ def clean(self):
                 }
             )
 
-        # Validate device software version has a software image file that matches the device's device type or is a default image
-        if self.software_version is not None and not any(
-            (
-                self.software_version.software_image_files.filter(device_types=self.device_type).exists(),
-                self.software_version.software_image_files.filter(default_image=True).exists(),
-            )
-        ):
-            raise ValidationError(
-                {
-                    "software_version": (
-                        f"No software image files for version '{self.software_version}' are "
-                        f"valid for device type {self.device_type}."
+        # If any software image file is specified, validate that
+        # each of the software image files belongs to the device's device type or is a default image
+
+        if self.software_image_files.exists():
+            for image_file in self.software_image_files.all():
```

Comment:
The `if` is probably unnecessary here since the `for` will just be a no-op if it's an empty queryset?